packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -546,28 +547,37 @@ std::variant<rocksdb::Status, MachineStateKeys> ArbCore::getCheckpointUsingGas(
     marshal_uint256_t(total_gas, key);
     auto key_slice = vecToSlice(key);
     it->SeekForPrev(key_slice);
-    if (!it->Valid()) {
-        if (!it->status().ok()) {
-            return it->status();
+    while (it->Valid()) {
+        if (after_gas) {
+            it->Next();
+            if (!it->status().ok()) {
+                return it->status();
+            }
+            if (!it->Valid()) {
+                return rocksdb::Status::NotFound();
+            }
         }
-        return rocksdb::Status::NotFound();
-    }
-    if (after_gas) {
-        it->Next();
         if (!it->status().ok()) {
             return it->status();
         }
```

Comment:
This check is unnecessary, as if the status is not ok, it'll exit the loop as the iterator isn't valid, and then return the status.

Reply:
gone :)